tests/test_transaction.py

```diff
@@ -322,6 +323,14 @@ def test_named_savepoint_empty_string_invalid(conn):
         conn.transaction(savepoint_name="")
 
 
+@pytest.mark.xfail(raises=OperationalError, reason="TODO: Escape sp names")
```

Comment:
I've added [`pq.Escaping.escape_literal()`](https://github.com/psycopg/psycopg3/commit/a379f256687029c1d475adaaec5e3683dd8e2a20) in master: you can use it to sanitise the savepoint name.

Reply:
I'll do this after I rebase the branch from master, but I want to close out this PR first as it's quite long and confusing, so I want to draw a line under it now without introducing more complexity with rebase/merge commits.

I've made a note of this on the todo list at  https://github.com/asqui/psycopg3/projects/1 so that this doesn't get forgotten.